### ISSUE TITLE
Regression: Nested Callable Protocol leads to reportUnknownArgumentType

### ISSUE THREAD

**Issue by pvandyken** [microsoft/pyright#5356]
Note: if you are reporting a wrong signature of a function or a class in the standard library, then the typeshed tracker is better suited for this report: https://github.com/python/typeshed/issues.

If you have a question about typing or a behavior that you’re seeing in Pyright (as opposed to a bug report or enhancement request), consider posting to the [Pyright discussion forum](https://github.com/microsoft/pyright/discussions).

**Describe the bug**
In pyright versions 1.1.312 through 1.1.315, using a callable protocol's return directly as an argument to the same protocol results in a `reportUnknownArgumentType` error, if the protocol's `__call__` uses a `TypeVar`, as shown in the example below. I noticed a number of bug fixes related to generic functions in the `1.1.312` release notes, but perhaps this is a case that was missed?

**To Reproduce**
python v3.9.7
pyright v1.1.315

Calling pyright on the following file:
```py
from __future__ import annotations
from typing import Any, Protocol, TypeVar

T = TypeVar("T")

class Prot(Protocol):
    def __call__(self, func: T) -> T:
        ...

def example(
    func: Prot,
    names: Any,
):
    func(func(names))
		#~~~~~~~~~~~
		#Argument Type is Unknown
```

```
/.../test.py
  /.../test.py:14:10 - error: Argument type is unknown
  Argument corresponds to parameter "func" in function "__call__" (reportUnknownArgumentType)
1 error, 0 warnings, 0 informations
```

**Expected behavior**
I would expect the above example to infer correctly, as it does in pyright<1.1.312

**VS Code extension or command-line**
Error occurs in the pip installed pyright library and in the vscode extension

**Additional context**

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by pvandyken** [comment]
Thanks so much!

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.316, which I just published. It will also be included in this week's insiders build of pylance.
